cpp/src/sort/sort_column_impl.cuh

```diff
@@ -0,0 +1,160 @@
+  template <typename T>
+  void faster_sort(column_view const& input,
+                   mutable_column_view& indices,
+                   bool ascending,
+                   rmm::cuda_stream_view stream)
+  {
+    // A thrust sort on a column of primitive types will use a radix sort.
+    // For other fixed-width types, thrust will use merge-sort.
+    // But this also requires making a copy of the input data.
+    auto temp_col = column(input, stream);
+    auto d_col    = temp_col.mutable_view();
+    if (ascending) {
+      if constexpr (stable) {
+        thrust::stable_sort_by_key(rmm::exec_policy(stream),
+                                   d_col.begin<T>(),
+                                   d_col.end<T>(),
+                                   indices.begin<size_type>(),
+                                   thrust::less<T>());
+      } else {
+        thrust::sort_by_key(rmm::exec_policy(stream),
+                            d_col.begin<T>(),
+                            d_col.end<T>(),
+                            indices.begin<size_type>(),
+                            thrust::less<T>());
+      }
+    } else {
+      if constexpr (stable) {
+        thrust::stable_sort_by_key(rmm::exec_policy(stream),
+                                   d_col.begin<T>(),
+                                   d_col.end<T>(),
+                                   indices.begin<size_type>(),
+                                   thrust::greater<T>());
+      } else {
+        thrust::sort_by_key(rmm::exec_policy(stream),
+                            d_col.begin<T>(),
+                            d_col.end<T>(),
+                            indices.begin<size_type>(),
+                            thrust::greater<T>());
+      }
+    }
+  }
+
+  /**
+   * @brief Sorts a single column with a relationally comparable type.
+   *
+   * This is used when a comparator is required.
+   *
+   * @param input Column to sort
+   * @param indices Output sorted indices
+   * @param ascending True if sort order is ascending
+   * @param null_precedence How null rows are to be ordered
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   */
+  template <typename T>
+  void sorted_order(column_view const& input,
+                    mutable_column_view& indices,
+                    bool ascending,
+                    null_order null_precedence,
+                    rmm::cuda_stream_view stream)
+  {
+    auto keys = column_device_view::create(input, stream);
+    auto comp = simple_comparator<T>{*keys, input.has_nulls(), ascending, null_precedence};
+    if constexpr (stable) {
+      thrust::stable_sort(
+        rmm::exec_policy(stream), indices.begin<size_type>(), indices.end<size_type>(), comp);
+    } else {
+      thrust::sort(
+        rmm::exec_policy(stream), indices.begin<size_type>(), indices.end<size_type>(), comp);
+    }
+  }
+
+  template <typename T,
+            std::enable_if_t<cudf::is_relationally_comparable<T, T>() and
```

Comment:
I would recommend trying `CUDF_ENABLE_IF` (it can be used here since we don't have device lambda).